src/escape.rs

```diff
@@ -86,30 +101,60 @@ static HTML_ESCAPE_TABLE: [u8; 256] = [
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     ];
 
-static HTML_ESCAPES: [&'static str; 6] = [
+static HTML_ESCAPES: [&'static str; 7] = [
         "",
         "&quot;",
         "&amp;",
         "&#47;",
         "&lt;",
-        "&gt;"
+        "&gt;",
+        "\\",
     ];
 
-pub fn escape_html(ob: &mut String, s: &str, secure: bool) {
+/// What does secure mean?
+pub fn escape_html(ob: &mut String, s: &str, skip_unescape: bool) {
     let size = s.len();
     let bytes = s.as_bytes();
     let mut mark = 0;
     let mut i = 0;
+
     while i < size {
         match bytes[i..].iter().position(|&c| HTML_ESCAPE_TABLE[c as usize] != 0) {
             Some(pos) => {
                 i += pos;
             }
             None => break
         }
-        let c = bytes[i];
+        if !skip_unescape {
```

Comment:
The input `&lt;` should be in the AST as `Text("<")`. I'm about 90% sure there doesn't need to be special case escaping here for code blocks. Of course, when *parsing* the rules are different - if the input `&lt;` is in a code block, then the AST holds `Text("&lt;")`.